p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/ProjectDependencyClosureGraph.java

```diff
@@ -55,88 +71,166 @@ class ProjectDependencyClosureGraph implements ProjectDependencyClosure {
 
 	private Map<MavenProject, ProjectDependencies> projectDependenciesMap;
 
-	private InstallableUnitSlicer slicer;
-
 	private Map<MavenProject, Collection<IInstallableUnit>> projectIUMap;
 
-	ProjectDependencyClosureGraph(Map<MavenProject, Collection<IInstallableUnit>> projectIUMap,
-			InstallableUnitSlicer slicer) throws CoreException {
+	// Graph structure: maps each project to its outgoing edges
+	private Map<MavenProject, Set<Edge>> projectEdgesMap = new HashMap<>();
```

Comment:
```suggestion
	private Map<MavenProject, List<Edge>> projectEdgesMap = new HashMap<>();
```

As we not need lockup edges it will prevent comparison for duplicates.